xtask/src/ai.rs

```diff
@@ -0,0 +1,129 @@
+use clap::Parser;
+use nu_ansi_term::Color::{Blue, DarkGray, Green, LightGray};
+use reedline::FileBackedHistory;
+use reedline::{DefaultPrompt, DefaultPromptSegment, Reedline, Signal};
+use std::env;
+use test_context::AsyncTestContext;
+use trustify_module_fundamental::ai::model::{ChatState, MessageType};
+use trustify_module_fundamental::ai::service::AiService;
+use trustify_test_context::TrustifyContext;
+
+#[derive(Debug, Parser, Default)]
+pub struct Ai {}
+
+impl Ai {
+    pub async fn run(self) -> anyhow::Result<()> {
+        run().await
+    }
+}
+
+async fn run() -> anyhow::Result<()> {
+    // Set the following environment variables to against a different OpenAI API endpoint
+    // OPENAI_API_BASE, OPENAI_API_KEY, OPENAI_MODEL
+
+    // export EXTERNAL_TEST_DB to use an external database, otherwise
+    // a temporary database will be created for testing.
+    let ctx = TrustifyContext::setup().await;
+
+    // if we are not using an external database, ingest some documents so that
+    // we use the data in them to respond to questions.
+    if env::var("EXTERNAL_TEST_DB").is_err() {
+        println!("{}", LightGray.paint("ingesting documents..."));
+        _ = ctx
```

Comment:
I think the `?` counts as using the result enough you don't have to assign to `_`

Reply:
Your right... simplified now.